libkmod/libkmod-module.c

```diff
@@ -1072,17 +1072,15 @@ KMOD_EXPORT int kmod_module_probe_insert_module(struct kmod_module *mod,
 			return 0;
 	}
 
-	/*
-	 * Ugly assignment + check. We need to check if we were told to check
-	 * blacklist and also return the reason why we failed.
-	 * KMOD_PROBE_APPLY_BLACKLIST_ALIAS_ONLY will take effect only if the
-	 * module is an alias, so we also need to check it
-	 */
-	if ((mod->alias != NULL && ((err = flags & KMOD_PROBE_APPLY_BLACKLIST_ALIAS_ONLY)))
-			|| (err = flags & KMOD_PROBE_APPLY_BLACKLIST_ALL)
-			|| (err = flags & KMOD_PROBE_APPLY_BLACKLIST)) {
-		if (module_is_blacklisted(mod))
-			return err;
+	if (module_is_blacklisted(mod)) {
```

Comment:
I'd OR all flags together and check that before checking if module is blacklisted.

Reply:
Reconsidered: one of these flags are actually **likely** to be set so it doesn't matter checking for the flag before checking if the module is blacklisted. Example: src/shared/module-util.c in systemd.